From 9338df316ef30067f846b0482d856bd7914b9af6 Mon Sep 17 00:00:00 2001
From: Jason Lowe <jlowe@apache.org>
Date: Wed, 3 Aug 2016 19:17:25 +0000
Subject: [PATCH 1991/2748] YARN-5462.
 TestNodeStatusUpdater.testNodeStatusUpdaterRetryAndNMShutdown
 fails intermittently. Contributed by Eric Badger
 (cherry picked from commit
 db646540f094077941b56ed681a4f3e5853f5b7f)

(cherry picked from commit 1ea03411c2568471a903ff004625052586aa264c)

Change-Id: I979acd6766842349df8de69dba747de2a54dee06
---
 .../server/nodemanager/TestNodeStatusUpdater.java  |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/TestNodeStatusUpdater.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/TestNodeStatusUpdater.java
index aca5511..f4034f0 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/TestNodeStatusUpdater.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/test/java/org/apache/hadoop/yarn/server/nodemanager/TestNodeStatusUpdater.java
@@ -474,6 +474,8 @@ protected NodeStatusUpdater createNodeStatusUpdater(Context context,
 
     @Override
     protected void serviceStop() throws Exception {
+      // Make sure that all containers are started before starting shutdown
+      syncBarrier.await(10000, TimeUnit.MILLISECONDS);
       System.out.println("Called stooppppp");
       super.serviceStop();
       isStopped = true;
@@ -1452,6 +1454,9 @@ public void testNodeStatusUpdaterRetryAndNMShutdown()
       new File("start_file.txt"));
 
     try {
+      // Wait until we start stopping
+      syncBarrier.await(10000, TimeUnit.MILLISECONDS);
+      // Wait until we finish stopping
       syncBarrier.await(10000, TimeUnit.MILLISECONDS);
     } catch (Exception e) {
     }
-- 
1.7.9.5

